src/lobby/username.ts

```diff
@@ -0,0 +1,24 @@
+import { setup } from '../setup/internal'
+
+import { USERNAME_BLOCKLIST } from './blocklist'
+
+/**
+ * Check if a username is available.
+ */
+export async function isUsernameAvailable(
+  username: string
+): Promise<boolean> {
+  const resp = await fetch(`${setup.endpoints.api}/user/data/${username}`)
+  return !resp.ok
+}
+
+/**
+ * Check if a username is valid.
+ */
+export function isUsernameValid(username: string): boolean {
+  return !username.startsWith("-") &&
+         !username.endsWith("-") &&
+         !username.startsWith("_") &&
+         /^[a-zA-Z1-9_-]*$/.test(username) &&
```

Comment:
I think the `*` has to be a `+` here, so the string can't be empty.

Reply:
Good looks :+1: